sources/views/data.js

```diff
@@ -1,11 +1,93 @@
 import {JetView} from "webix-jet";
-import {data} from "models/records";
+import {countries} from "models/countries";
+import {statuses} from "models/statuses";
+import {useful_functions} from "models/useful";
 
 export default class DataView extends JetView{
 	config(){
-		return { view:"datatable", autoConfig:true, css:"webix_shadow_medium" };
+
+		var tabview = {
+	    view:"tabview",
+	    cells:[
+	      {
+	        header:"Countries",
+	        body:this.tab_datatable_element("countries_datatable", "Country")
```

Comment:
the principle is correct, but not the implementation. 

you need to 

- create an abstract GridView class with the `config` method that returns the configuration like in your `tab_datatable_element`
- create 2 more classes CountriesView and StatusesView that extend from this GridView, not Jetview
-  classes CountriesView and StatusesView should have init methods and load countries or statuses depending on the variable in their constructor
- classes CountriesView and StatusesView should be embedded into DataView as `body: new StatusesView(this.app, "", statuses)`

study slides 28 and 29 https://slides.com/ihelga/webixjet-views#/28 and the related demo to get the idea

Reply:
I have found a better solution. Instead of 3 classes, I have used 1 for creating 2 components

Reply:
looks ok as well. still, some issues need to be fixed:

(1) do not pass widget ID as parameter, this module does not need it. moreover, using webix global ids are not allowed in Webix Jet

(1) instead of loading data in config as `data:this.data`, place it into the init function

~~~
init(view){
    this.grid = view.queryView("datatable"); // here you search for the datatable
    this.grid.parse(this.data); // here you load the data
}
~~~ 

and use `this.grid` in a button click handler to add new items (instead of global `component_id`) and scroll

(3) `.add({Name:component_name})` - you can add an empty name here

Reply:
This methods don't work with "this.grid" (undefined in config()):
this.grid.add({Name:""});
scrollToLastAddedElemen(this.grid,true,true);

Reply:
you use `click:function(){ /* here this points to button*/}` 
but you can use arrow function to get to Jet class context `click():=>{ /*here this point to JetView class*/ }`

Reply:
Ok